### PR TITLE
feat(eslint-plugin): add support for nested namespaces to unsafe-member-access

### LINKED ISSUE #3292: [no-unsafe-member-access] Not allowing "interface X extends" for a type in a nested namespace
- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md) and my problem is not listed.

**Repro**

[Playground](https://typescript-eslint.io/play#ts=5.3.2&fileType=.tsx&code=CYUwxgNghgTiAEA7KBbEBnADlMCAaB8A3gFDzwCWiALiDAGY4IAqIKmAjMWefPQPb8AXPABEAI1iieAXx7I0WJvADqa7r0o06jXPFbsATBs19BIiVJ7k5NknJJVaDZa3TVjIAB61EwdPAEeAB0airBBpjGRPByQA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Jge1tieQEMAZoloBbRCIBGiaLT5kyKVBkjToHaJHBgAviG1A&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eFYDAruuGAL4g9A&tokens=false)

I wasn't able to build a repro for this issue, but it really seems like a bug.

I have the following `d.ts` file on my project (I'm using namespaces to organize my types).

**XXX.d.ts**

```
declare namespace XXX {

  interface Temp1 {
    foo: "bar"
  }

  namespace WWW {

    interface Temp2 {
      foo: "bar"
    }

  }

}
```

So the `XXX` is ambient and the `WWW` namespace is nested within it.

I need to extend the `Temp2` interface, that lives on the nested `WWW` namespace.

So I did this:

```
interface Test2 extends XXX.WWW.Temp2 {}
```

And I got this error:

image

So I thought: "eslint might not be detecting the XXX namespace". But that is not true, as you can see that the following code works fine when extending `Temp1` which is a direct member of the XXX namespace.

```
interface Test1 extends XXX.Temp1 {}
```

image

And what is even weirder is that if I use an intermediate `AUX` type, it works fine:

```
type AUX = XXX.WWW.Temp2;
interface Test2A extends AUX {}
```

image

**Expected Result**

There shouldn't be any errors when extending a nested namespace.

```
interface Test2 extends XXX.WWW.Temp2 {}
```

**Actual Result**

The `@typescript-eslint/no-unsafe-member-access` is being triggered.

**Additional Info**

**Versions**

| package                     | version |
| --------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `4.22.0` |
| `@typescript-eslint/parser` | `4.22.0` |
| `TypeScript`                | `4.2.4` |
| `ESLint`                    | `7.24.0` |
| `node`                      | `10.19.0` |

**.eslintrc.js** 

```
const config = {

  root: true,
  

  env: {
    browser : true,
    commonjs: true,
    es2017: true,
    node: true
  },
  parser: "@typescript-eslint/parser",
  ignorePatterns: [
    ".eslintrc.js",
    "webpack.*.ts",
    "babel.config.*.js"
  ],
  parserOptions: {
    project: "./tsconfig.json",
    sourceType: "module",
  },

  globals: {
    gtag: "readonly"
  },

  extends: [
    "eslint:recommended",
    "plugin:react/recommended",
    "plugin:react-hooks/recommended",
    "plugin:@typescript-eslint/recommended",
    "plugin:@typescript-eslint/recommended-requiring-type-checking"
  ],

  plugins:[
    "react",
    "react-hooks",
    "@typescript-eslint"
  ],

  rules: {
    "semi": "error",
    "react/prop-types": "off",
    "react-hooks/rules-of-hooks": "error",
    "react-hooks/exhaustive-deps": "warn",
    "@typescript-eslint/no-unused-vars": ["warn", {"args":"none"}],
    "@typescript-eslint/no-empty-interface": "off",
    "@typescript-eslint/prefer-regexp-exec": "off",
    "@typescript-eslint/require-await": "off"
  },
  
  settings: {
    "react": {
      version: "detect"
    }
  },

  overrides: [
    { 
      files: ["*.js"],
      rules: {
        "@typescript-eslint/no-var-requires": 0
      }
    }
  ]
};

module.exports = config;
```

**tsconfig.json**

```
{
  "compilerOptions": {
    "target": "es5",
    "lib": [
      "dom",
      "dom.iterable",
      "esnext"
    ],
    "allowJs": true,
    "skipLibCheck": true,
    "strict": false,
    "strictNullChecks": true,
    "forceConsistentCasingInFileNames": true,
    "noEmit": true,
    "esModuleInterop": true,
    "module": "esnext",
    "moduleResolution": "node",
    "resolveJsonModule": true,
    "isolatedModules": true,
    "jsx": "preserve",
    "types": ["node"],
    "baseUrl": ".",
    "paths": {
      "@app/*": ["./app/*"],
      "@lib/*": ["./lib/*"]
    }
  },
  "include": [
    "next-env.d.ts",
    "**/*.ts",
    "**/*.tsx"
  ],
  "exclude": [
    "node_modules"
  ]
}

```

### PR DESCRIPTION
**Pull request by JSanchezIO**
## PR Checklist

- [x] Addresses an existing open issue: fixes #3292 
- [x] That issue was marked as [accepting prs](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aopen+is%3Aissue+label%3A%22accepting+prs%22)
- [x] Steps in [Contributing](https://typescript-eslint.io/contributing) were taken

## Overview

The rule `no-unsafe-member-access` was originally targeted all `MemberExpression` nodes. Eventually, it was updated, #2753, to ignore a `MemberExpression` if its parent was either `TSClassImplements` or `TSInterfaceHeritage`. However, this only accounts for one-level of nesting in namespaces. This change accounts for any level of nesting by ignoring the `MemberExpression` if the root node is `TSClassImplements` or `TSInterfaceHeritage`.

### CODE REVIEW

**Review by auvred — DISMISSED** [review]
Awesome! :+1:

File: packages/eslint-plugin/src/rules/no-unsafe-member-access.ts [hunk 1]
```diff
@@ -107,8 +107,8 @@ export default createRule({
     }
 
     return {
-      // ignore MemberExpression if it's parent is TSClassImplements or TSInterfaceHeritage
-      ':not(TSClassImplements, TSInterfaceHeritage) > MemberExpression':
+      // ignore MemberExpression's with ancestor's of type `TSClassImplements` or `TSInterfaceHeritage`
```
Comment (JoshuaKGoldberg) [inline packages/eslint-plugin/src/rules/no-unsafe-member-access.ts:110]: [Docs] Grammar nit: these are plurals, not possessives:

```suggestion
      // ignore MemberExpressions with ancestors of type `TSClassImplements` or `TSInterfaceHeritage`
```

**Review by JoshuaKGoldberg — APPROVED** [review]
🚀

**Review by JoshuaKGoldberg — APPROVED** [review]
🚀 thanks!

**Pull request merged**